plugin/Model/FormModelBuilder/FormModelBuilder.php

```diff
@@ -130,19 +137,22 @@ protected function readTemplatePaths($directory, $pattern = '*')
 			$file = substr(strrchr($value, "/"), 1);
 			if(strpos($value, 'themes') !== false) {
 				$value = __String::getNew($value)->replace(get_template_directory().'/', '');
-				$templates[1]['path'][$value] = $file;
-				$templates[1]['title'] = 'Personalized (Theme)';
-				$templates[1]['folder'] = '/onoffice-theme/templates/' . $directory . '/';
+				$folderOrder = $this->getOrderOfTemplateFolder('theme');
+				$templates[$folderOrder]['path'][$value] = $file;
+				$templates[$folderOrder]['title'] = 'Personalized (Theme)';
+				$templates[$folderOrder]['folder'] = '/onoffice-theme/templates/' . $directory . '/';
 			}else{
 				$value = __String::getNew($value)->replace(plugin_dir_path(ONOFFICE_PLUGIN_DIR), '');
 				if (strpos($value, 'onoffice-personalized') !== false) {
-					$templates[2]['path'][$value] = $file;
-					$templates[2]['title'] = 'Personalized (Plugin)';
-					$templates[2]['folder'] = 'onoffice-personalized/templates/' . $directory . '/';
+					$folderOrder = $this->getOrderOfTemplateFolder('plugin');
+					$templates[$folderOrder]['path'][$value] = $file;
```

Comment:
one `$templates[$folderOrder]` can only have one template?

Reply:
@jayay I think not, `$templates` is an array to store templates folder, so `$templates[$folderOrder]` will include information about that folder and also its templates, I think it's fine to use this way, if you have any good method more, please describe to me.

Reply:
Guess I didn't understand `$templates[$folderOrder]['path'][$value] = $file`.

Do you think a data model would be easier to deal with here?

Reply:
I think it doesn't make much difference